internal/provider/kubernetes/gateway.go

```diff
@@ -65,6 +68,11 @@ func newGatewayController(mgr manager.Manager, cfg *config.Server, su status.Upd
 	}
 	r.log.Info("watching gateway objects")
 
+	// Watch Services owned by managed Gateways.
+	if err := c.Watch(&source.Kind{Type: &corev1.Service{}}, r.enqueueRequestForOwningGateway()); err != nil {
```

Comment:
stared at this line for a while to understand why its needed - "trigger translation as well as updating the gateway status when the envoy service managed by the infra manager has changed". Would be great to add a similar comment in the code to make it easier for the next dev